src/blockchain/mod.rs

```diff
@@ -84,6 +84,32 @@ pub enum Capability {
     AccurateFees,
 }
 
+/// The height and confirmation status of a transaction
+pub struct TxStatus {
+    /// Whether the transaction has at least one confirmation
+    pub confirmed: bool,
+    /// The height of the block the transaction was confirmed in
+    pub block_height: Option<u32>,
+}
+
+/// A trait Blockchains can implement if they support querying chain data
```

Comment:
By "support querying chain data" do you mean availability of certain indexing in those blockchain? In that case it might make sense to elaborate the doc a little to explain what are those special requirements. Also how `IndexedChain` different from regular backends with the `Blockchain` trait.

In general a little explainer doc for this new trait would be very nice.